src/LuaTransformer.ts

```diff
@@ -2111,23 +2114,33 @@ export class LuaTransformer {
     }
 
     public transformReturnStatement(statement: ts.ReturnStatement): StatementVisitResult {
+        // Bubble up explicit return flag and check if we're inside a try/catch block
+        let insideTryCatch = false;
+        for (let i = this.scopeStack.length - 1; i >= 0; --i) {
```

Comment:
Can insideTryCatch be set to false after being true again? Otherwise you could also add `!insideTryCatch` to the loop condition here.

Reply:
If the loop exits on the first `try`/`catch` it hits, the `functionReturned` flag won't bubble up to nested try/catch statements.